build-support/native-image/utils.bash

```diff
@@ -0,0 +1,57 @@
+# Functions used in building zinc native-images.
+# TODO: This will be made automatic in pants via https://github.com/pantsbuild/pants/pull/6893.
+
+export NATIVE_IMAGE_BUILD_CACHE_DIR="${HOME}/.cache/pants/native-image-build-script-cache"
+
+function is_osx {
+  [[ "$(uname)" == 'Darwin' ]]
+}
+
+function with_pushd {
+  pushd "$1" >&2
+  "${@:2}"; rc="$?"
+  popd >&2
+  return "$rc"
+}
+
+function do_within_cache_dir {
+  with_pushd "$NATIVE_IMAGE_BUILD_CACHE_DIR" "$@"
+}
+
+function normalize_path {
+  realpath "$@" \
```

Comment:
AFAIK OSX doesn't come with `realpath` or support `readlink -f` by default

Reply:
I have attempted to fix this by:
- checking if `realpath` is available
- if not, checking if `brew` is available
- if so, installing coreutils, otherwise printing an error message pointing to https://brew.sh.